Removed DF Hotkey and fixed module.json

tehguitarist opened the pull request:
This pull should get everything functional again. I don't think it fixes the inability to move the hotbar, but CSS isn't my wheelhouse (programming barely is ha). Basically it just removes all code that references DF Hotkeys.

I fixed up the deprecation warnings for the module.json, I wasn't sure if colorsettings was explicitly required or just a nice to have, so i've left it as optional, but that can be changed to required easily enough.

I bumped the version number and links to avoid any confusion (and so my version doesn't autoupdate by accident).

I didn't delete the hotkey code, just commented out, as I figured if you wanted to transpose that to the core keybinding system, it would be safest not to delete it.

I also added a release zip just for anyone who needs a cleaner download to use in their own worlds.

Commit:
Removed DF Hotkey and fixed module.json

Commit:
Added release zip

charlesfire commented:
`I didn't delete the hotkey code, just commented out, as I figured if you wanted to transpose that to the core keybinding system, it would be safest not to delete it.`

No need to keep commented code. If needed, the removed code can be easily retrieved from the history. It's kind of the point of git, to keep track of changes.

tehguitarist commented:
Didn’t think of that. If I find some time I’ll clean everything up if Norc doesn’t merge this soonish.

Norc commented:
Hey, thanks so much for the pull request!! I'll take a look at this and potentially merge tonight. 

It is greatly, greatly appreciated.

Commit:
cleaned up commented code

tehguitarist commented:
Cleaned up the commented code and added TODO notes instead. Mind you, JS and CSS are so far out of my wheel house, tinkering is about as good as I get ha.

Commit:
Fix for keyDiv warning & DorakoUI hotbar position

tehguitarist commented:
Fixed an error warning around a keyhintDiv error and added in some CSS to fix the issue of PF2e Dorako UI stopping the hotbar being able to shift position. Also added an adjustment to handle Dorako UI sizing things a little different so the default position works nice.

If other UI modules are breaking things, it's possible similar code might fix it too.

Commit:
Verified it still works in V11

tehguitarist commented:
Just did a test, seems to still work in V11, so updated the module.json to reflect that.

tehguitarist commented:
Hey, just thought I'd follow up to see if you needed any more changes.

GhostwheelX commented:
What .json should I add in Foundry to use this version?

tehguitarist commented:
> What .json should I add in Foundry to use this version?

I haven’t set this up as a manifest you can just add, but if you replace the files with the ones from my fork, that’ll do the trick

GhostwheelX commented:
I did so, and when I hit "update all", I receive the following error:
![image](https://github.com/Norc/foundry-custom-hotbar/assets/68725723/85bcf2a6-dc5a-4d98-8455-9f3ecf14e336)

Is this something that will resolve itself when the main module is updated, or do I need to keep an eye out for when it's updated and then uninstall and re-install?

tehguitarist commented:
Yeah when the module gets updated that should resolve itself. I did it because the module would sometimes “update” itself to the main version. 
I’ve since just locked it so it doesn’t update. It’s a harmless warning though. I’ll try to maintain this as best I can, but I’m hesitant to maintain it until I see what the v12 macro changes are. I’m not a professional by any stretch of the imagination so if there are fundamental changes to macros coming, I doubt I could port them over.

Arkandos commented:
I tried updating manually with this fork on foundry v11, but it removes the custom hotbar entirely, and makes the normal hotbar inaccessible. 
![image](https://github.com/user-attachments/assets/c075890f-0002-4000-9677-23aec87f9067)
When its the only module enabled, it also shifts all other parts of the ui upwards, making them inaccessible 
![image](https://github.com/user-attachments/assets/fab1ec4c-fb73-44ea-bab0-045d466faaa7)

@tehguitarist
